python/cuspatial/cuspatial/core/spatial/distance.py

```diff
@@ -480,6 +481,92 @@ def pairwise_point_polygon_distance(points: GeoSeries, polygons: GeoSeries):
     )
 
 
+def pairwise_linestring_polygon_distance(
+    linestrings: GeoSeries, polygons: GeoSeries
+):
+    """Compute distance between pairs of (multi)linestrings and (multi)polygons
+
+    The distance between a (multi)linestrings and a (multi)polygon
+    is defined as the shortest distance between every segment in the
+    multilinestring and every edge of the (multi)polygon. If the
+    multilinestring and multipolygon intersects, the distance is 0.
+
+    This algorithm computes distance pairwise. The ith row in the result is
+    the distance between the ith (multi)linestring in `linestrings` and the ith
+    (multi)polygon in `polygons`.
+
+    Parameters
+    ----------
+    linestrings : GeoSeries
+        The (multi)linestrings to compute the distance from.
+    polygons : GeoSeries
+        The (multi)polygons to compute the distance from.
+
+    Returns
+    -------
+    distance : cudf.Series
+
+    Notes
+    -----
+    The input `GeoSeries` must contain a single type geometry.
+    For example, `linestrings` series cannot contain both linestrings and
+    polygons.
+
+    Examples
+    --------
+    Compute distance between a linestring and a polygon:
+    >>> from shapely.geometry import LineString, Polygon
+    >>> lines = cuspatial.GeoSeries([
+    ...     LineString([(0, 0), (1, 1)])])
+    >>> polys = cuspatial.GeoSeries([
+    ...     Polygon([(-1, -1), (-1, 0), (-2, 0), (-1, -1)])
+    ... ])
+    >>> cuspatial.pairwise_linestring_polygon_distance(lines, polys)
+    0    1.0
+    dtype: float64
+
+    Compute distance between a multipoint and a multipolygon
+    >>> from shapely.geometry import MultiLineString, MultiPolygon
+    >>> lines = cuspatial.GeoSeries([
+    ...     MultiLineString([
+    ...             LineString([(0, 0), (1, 1)]),
+    ...             LineString([(1, 1), (2, 2)])])
+    ...     ])
+    >>> polys = cuspatial.GeoSeries([
+    ...     MultiPolygon([
+    ...             Polygon([(-1, -1), (-1, 0), (-2, 0), (-1, -1)]),
+    ...             Polygon([(-2, 0), (-3, 0), (-3, -1), (-2, 0)])])
+    ...     ])
+    >>> cuspatial.pairwise_linestring_polygon_distance(lines, polys)
+    0    1.0
+    dtype: float64
+    """
+
+    if len(linestrings) != len(polygons):
+        raise ValueError("Unmatched input geoseries length.")
+
+    if len(linestrings) == 0:
+        return cudf.Series(dtype=linestrings.lines.xy.dtype)
+
+    if not contains_only_linestrings(linestrings):
+        raise ValueError("`linestrings` array must contain only linestrings")
+
+    if not contains_only_polygons(polygons):
+        raise ValueError("`polygon` array must contain only polygons")
+
+    # Handle slicing in geoseries
+    linestrings_column = linestrings._column.lines._column.take(
+        linestrings._column._meta.union_offsets._column
+    )
+    polygon_column = polygons._column.polygons._column.take(
+        polygons._column._meta.union_offsets._column
+    )
```

Comment:
I guess this is reducing the `GeoColumn` into a pure `ListColumn` here? It seems like an accessor would improve this here and elsewhere. Perhaps a simple modification to the `GeoColumnAccessor` to return this or similar would improve readability.

Reply:
That's correct. Something like this might already exists. I'll look into the accessors.

Reply:
I added such accessor and refactors this code.